cve_bin_tool/input_engine.py

```diff
@@ -115,24 +115,21 @@ def strip_remark(detail) -> str:
                         ref = affect["ref"]
                         version = None
                         vendor = None
+                        # Expected ref format is urn:cbt:{bom_version}/{vendor}#{product}-{version}
                         if "#" in ref:
                             # Extract product information after # delimiter
                             p = ref.split("#")[1]
                             # Last element is version, rest is product which may include -
                             elements = p.rsplit("-", 1)
                             product = elements[0]
                             version = elements[1]
-                            # Now find vendor
-                            vendor_package_pair = self.cvedb.get_vendor_product_pairs(
-                                product
-                            )
-                            if vendor_package_pair != []:
-                                # FIXME https://github.com/intel/cve-bin-tool/issues/2320
-                                vendor = vendor_package_pair[
-                                    len(vendor_package_pair) - 1
-                                ]["vendor"]
+                            # Extract vendor between / and #
+                            tempstring = ref.split("/")
+                            if len(tempstring) > 1:
+                                vendor = ref.split("/")[1].split("#")[0]
```

Comment:
Okay,  this match clearly isn't working as planned.  Let me take a stab at getting it to behave.

```suggestion
                           # string will look like "ref": "urn:cbt:1/gnu#glibc-2.33"
                            vendor_match = re.match("[^/]*/([^#]*)#", ref)
                            if (vendor_match is not None):
                                vendor = vendor_match.group(1)
```